### ISSUE TITLE
Analytics

### ISSUE THREAD

**Issue by KraigWalker** [ncase/coming-out-simulator-2014#7]
@anubiann00b suggested having a graph of what choices and stuff people make. An analytics package could be used to follow the "flows" of what people choose, and what people choose most often.

I guess that sort of data can be used to figure out how people go into it. Do people make answers as themselves, or do they veer more on trying to get to the... end... as bluntly as possible.

Google Analytics is the old faithful, but Parse and Mixpanel are gorgeous, albeit under a free tier.

**Comment by ncase** [comment]
Yeah! I saw their suggestion before the game jam deadline! And I was planning to have a global stats screen at the end, similar to The Walking Dead's games... but then, well, deadline. I'll ask, but hopefully the jam organizer won't mind.

There is a global variable called "$" that contains the vars for all the important choices made and remembered! Could serialize & save those in any database, really. Maybe even Firebase, since that requires no server of our own?

**Comment by sylvia43** [comment]
I don't have much experience here; I just think analytics would be awesome.

**Comment by KraigWalker** [comment]
While it would be straightforward to have something like firebase, and push everybody's save games to a database, that would in fact be rather inefficient. Even though there's a vast range of outcomes and paths, we'd be likely to get more duplicate savegames than unique. The moment you've got a duplicate savegame, you've got one-too-many savegames. That's why I'd recommend something like [Parse](https://parse.com/products/analytics), [Google Analytics](http://www.google.com/analytics/), or [Mixpanel](https://mixpanel.com). You basically make calls out to their API when the player does something, like choose an option. These events are logged by the Analytics, and they handle the data side of things. I'm fairly green to Analytics myself. Only recently got GA going with my site and discovered they had a whole Events API.

**Comment by TheLastProject** [comment]
Please respect your visitor's privacy more than the Asian parents in the game. If you go with something Google Analytics-like, please at least go with Piwik ( http://piwik.org/ ).

Quite likely, though, serializing them into a local database would mean you could just grab them out and display them, which would be much more reliable and, in the "public domain" spirit, something everyone could easily use in their own version as well (assuming it is coded as an optional part of the game).

**Comment by sylvia43** [comment]
Yeah, definatly make it anonymous. What I was thinking is to send a packet every time a player makes a choice, and the server would record the number of times a choice is made, the time per choice, the number of times a game is started, the number of times, it's finished all the way, etc.

**Comment by ncase** [comment]
Yeah, I haven't used Google Analytics since... ever. I do actually use a self-hosted Piwik for the website of my main game project, Nothing To Hide! Or yeah, I could dump anonymized data to Firebase coz I'm a lazy loaf of laziness.

**Comment by sylvia43** [comment]
You're really hung on Firebase, aren't you.

**Comment by KraigWalker** [comment]
Hows about, rather than discussing a specific technical implementation (which may change) we discuss data points, and what data to present (and how).

 I don't know if you've seen anything Rockstar have done with their Social Club online service, but I really like the community stats page of collected factoids about the playerbase as a whole. Take for example [_L.A Noire's_](http://socialclub.rockstargames.com/games/lan/community) (alas, requires Social Club login) That's the kind of stat's page I have in my mind.
![LA Norie Community Stats Page Screenshot](https://cloud.githubusercontent.com/assets/1294877/3562274/c27b6896-09e7-11e4-96da-1cb9dbca19f8.jpg)

I think all of the data is pretty much anonymized anyways... Unless there's a giant Login with Facebook feature I completely missed out on.

There are some data points That may be interesting which aren't necessarily tied to the savegame in $. For example - how many people **don't** finish playing the game? I guess, there's a lot more dev related points like what browser/device/screen which probably wouldn't fit entirely right with a player community stats page, but good to know from the dev perspective.

**Comment by sylvia43** [comment]
Just send a packet every time an action is taken and keep count of which ones happen, rather than dumping whole game records.
